### PR TITLE
Newsletter signup improvement

### LINKED ISSUE #1676: Add error handling for newsletter subscription failure
We currently have the subscription form on 2 places on the site - one on homepage and one on /community. [We don't have error handling for server or network error in place yet](https://github.com/mozilla/teach.mozilla.org/blob/77465e206fd5b1b89df2de1c9d5fc24d6d37c03e/components/newsletter-signup/SignupForm.jsx#L45-L56). This means there's no feedback to users upon form submission failure.

Possible causes for this kind of error could be
- network problem (no internet connection etc)
- ET/Basket is down
- other error response that Basket can issue (@alicoding plz help name a few here)

Would love to hear some suggestion from @hannahkane @kristinashu @sabrinang @Pomax .

### PR DESCRIPTION
**Pull request by mmmavis**
Fixes #1676

**Changes proposed in this pull request:**
- got rid of external validator library `validator` and now rely on HTML5 form validation for client side form validation
- display error message returned back from Basket (invalid email or a more generalized error message)

---

- To test invalid email entry:
  1. remove `type="email"` in `SignupForm.jsx` around line 81
  2. enter an invalid email address, e.g., helloexample.com
  3. submit the form
  4. see if error message appears

- To simulate other Basket errors
 1. change `newsletters`  in `SignupForm.jsx` around line 32 to something random
 2. submit the form
 3. see if error message appears

### CODE REVIEW

**Comment by mmmavis** [comment]
@Pomax mind reviewing? 😃

File: components/newsletter-signup/SignupForm.jsx [hunk 1]
```diff
@@ -42,25 +35,28 @@ var SignupForm = React.createClass({
...
+          self.setState( {validationErrorType: basketErrorType} );
         }
       });
 
   },
   renderValidationErrors: function() {
-    if (this.state.validationErrors.length) {
+    var errorType = this.state.validationErrorType;
+    if ( errorType ) {
+      var errorMsg = errorType === "email" ? "Please enter a valid email address." : "There's been a problem with our system. Please try again later.";
```
Comment (Pomax) [inline components/newsletter-signup/SignupForm.jsx]: small suggestion: 
```
var errorMsg = "There's been a problem with our system. Please try again later.";
if (errorType === "email") {
  errorMsg = "Please enter a valid email address";
}
```
Mostly to allow for an easy extension beyond the binary error case.

**Comment by Pomax** [comment]
R+ although I don't know if our "generic" error has quite the right phrasing.

File: components/newsletter-signup/SignupForm.jsx (hunk 1 above)
Comment (mmmavis) [inline components/newsletter-signup/SignupForm.jsx]: cool, I thought about it when I wrote this part of code. Guess I made the wrong choice 😉 Will fix now.

**Comment by mmmavis** [comment]
getting the same Travis errors seen in https://github.com/mozilla/learning.mozilla.org/pull/1925#issuecomment-232149693

**Comment by cadecairos** [comment]
If this PR is going to merge some time, it will need to be rebased and reopened against the new base branch (master)

**Comment by mmmavis** [comment]
kk will do today

**Comment by mmmavis** [comment]
I've been working on a Network Pulse feature today - unfortunately I won't have time for this PR until I've successfully implemented that feature. 😬

**Pull request closed without merging**